### ISSUE TITLE
http://help.jabref.org/en/CustomImports/

### ISSUE THREAD

**Issue by oscargus** [JabRef/help.jabref.org#3]
The code example is not valid (e.g. `Util.createNeutralId()`). Also, there are references to SourceForge.

Finally, are there any existing custom importers that we would like to add directly to JabRef?

**Comment by koppor** [comment]
Link to click: http://help.jabref.org/en/CustomImports/

If you have a better code example, just edit the page :smiley:

**Comment by oscargus** [comment]
That may happen eventually. :-) Just thought it was better to put it here as a reminder.

**Comment by oscargus** [comment]
OK, so I updated the example and it works. I was also thinking about adding this example to the source repository so that it can be used for testing. However, I have currently two problems:

1. Where should I add the source? I do not really want the source to be in a package and if I add a java-file to e.g. src/test/resources/net/sf/jabref/importer/fileformat (which may make sense) it complains when I remove the package line.
2. How should a test like this be executed? One would need to script javac -classpath JabRefXX.jar file.java and once that is done, one can add the resulting class-file and then run an import test.

**Comment by tobiasdiez** [comment]
Good idea to add it as a test. I think if you add the class directly under \src\test\java\net\sf\jabref\importer, then it will compile along with the rest and you can just access it as a normal class.

**Comment by oscargus** [comment]
That is somehow what I would like to avoid just to make sure that the importer is actually imported and not just part of JabRef. Although, it is indeed a good idea to add it to make sure the example is actually working. My purpose was rather (which wasn't clear, I agree) to test the custom import mechanism rather than that particular example.

**Comment by oscargus** [comment]
Is there a way to include e.g. source code? I just realized that there are five different version of the page and it would be nice to not copy and paste to every version...

**Comment by Siedlerchr** [comment]
@oscargus  While the post is from February, I can now tell you, that it is now possible to have code blocks, same way as here on github. The site now supports GFM:.
http://kramdown.gettalong.org/parser/gfm.html
